libs/community/tests/integration_tests/test_vertex_ai_search.py

```diff
@@ -129,3 +133,177 @@ def test_cloudpickle(spec: Optional[Dict]) -> None:
     serialized = cloudpickle.dumps(retriever)
     retriever_loaded = pickle.loads(serialized)
     assert retriever.model_dump() == retriever_loaded.model_dump()
+
+@pytest.mark.parametrize("engine_data_type, get_extractive_answers, segment_config, expected_spec", [
+    (0, True, None, {
+        "max_extractive_answer_count": 1
+    }),
+    (0, False, {"num_previous_segments": 1, "num_next_segments": 1}, {
+        "max_extractive_segment_count": 1,
+        "num_previous_segments": 1,
+        "num_next_segments": 1,
+        "return_extractive_segment_score": False
+    }),
+    (0, False, {"num_previous_segments": 2, "num_next_segments": 3}, {
+        "max_extractive_segment_count": 1,
+        "num_previous_segments": 2,
+        "num_next_segments": 3,
+        "return_extractive_segment_score": False
+    }),
+    (0, False, {"num_previous_segments": 3, "num_next_segments": 2}, {
+        "max_extractive_segment_count": 1,
+        "num_previous_segments": 3,
+        "num_next_segments": 2,
+        "return_extractive_segment_score": False
+    }),
+    (0, False, {"num_previous_segments": 3, "num_next_segments": 2, "return_extractive_segment_score": True}, {
+        "max_extractive_segment_count": 1,
+        "num_previous_segments": 3,
+        "num_next_segments": 2,
+        "return_extractive_segment_score": True
+    }),
+])
+@pytest.mark.extended
```

Comment:
this sounds more like a unit test to me, should we move it to unit tests maybe?

Reply:
Yes, I moved the unit tests to the corresponding folder and added integration tests that use segments. I tested everything locally, and the tests are working well. Thank you @lkuligin ! 
Let me know if you have any suggestions.